main.go

```diff
@@ -186,7 +186,10 @@ func mailHandler(peer smtpd.Peer, env smtpd.Envelope) error {
 		err := cmd.Run()
 		if err != nil {
 			cmdLogger.WithError(err).Error(stderr.String())
-			return nil
+			return smtpd.Error{
+				Code:    554,
+				Message: "external sender error, contact your SMTP administrator",
```

Comment:
The error message sounds very generic. Could you make it slightly more precise?
What about: "External command failed"